src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
```

Comment:
[CODE PERFORMANCE] You are filtering the `Animal.alive` array every time an animal dies. This operation has a time complexity of O(n). It would be more efficient to use the `Array.prototype.findIndex` method to find the index of the animal that needs to be removed, and then use the `Array.prototype.splice` method to remove it. This way, you would only iterate over the array once, resulting in a time complexity of O(n), but with a smaller constant factor.